src/helpers.ts

```diff
@@ -371,3 +371,17 @@ export function isActionTrigger(ev: KeyboardEvent): boolean {
             return false;
     }
 }
+
+/*
+ * Create a shallow copy of an object.
+ */
+export function copyShallow<T extends object>(obj: T): T {
+    return Object.assign({}, obj);
+}
+
+/**
+ * Create a deep copy of an object by serializing and deserializing it.
+ */
+export function copyDeep<T extends object>(obj: T): T {
+    return JSON.parse(JSON.stringify(obj));
```

Comment:
I'm still puzzled by JS not having an exposed structured clone function... I'll add it to my list of JS quirks, just next to *everything is a float, deal with it... oh, and forget about u64*.